File: src/tasks/cli/command/common/CommonParser.java
package tasks.cli.command.common;

import org.apache.commons.cli.CommandLine;
import tasks.cli.parser.ParserUtil;

public final class CommonParser {
  public CommonParser() {}

  public <T> CommonArguments<T> parse(CommandLine commandLine, T specificArguments) {
    return new CommonArguments<>(
        specificArguments,
        !ParserUtil.getFlagPresence(commandLine, StripColors.STRIP_COLORS_OPTION.value()));
  }
}

File: src/tasks/cli/command/common/CommonParser.kt
package tasks.cli.command.common

import org.apache.commons.cli.CommandLine
import tasks.cli.parser.ParserUtil

class CommonParser {
  fun <T> parse(commandLine: CommandLine, specificArguments: T): CommonArguments<T> {
    return CommonArguments(
        specificArguments,
        !ParserUtil.getFlagPresence(commandLine, StripColors.STRIP_COLORS_OPTION.value()))
  }
}